lib/FluxibleContext.js

```diff
@@ -190,46 +197,81 @@ FluxContext.prototype._initializeDispatcher = function initializeDispatcher() {
 /**
  * Returns the context for action controllers
  * @method getActionContext
+ * @param {Object} meta Metadata to keep track of the current flow of executing actions
  * @return {Object} Action context information
  */
-FluxContext.prototype.getActionContext = function getActionContext() {
-    if (this._actionContext) {
-        return this._actionContext;
-    }
+FluxContext.prototype.getActionContext = function getActionContext(meta) {
     var self = this;
 
-    if (!self._dispatcher) {
-        self._initializeDispatcher();
-    }
-
-    var actionContext = {
-        dispatch: this._dispatcher.dispatch.bind(this._dispatcher),
-        executeAction: this.executeAction.bind(this),
-        getStore: this._dispatcher.getStore.bind(this._dispatcher)
-    };
-
-    self._plugins.forEach(function pluginsEach(plugin) {
-        var actionContextPlugin = plugin.plugActionContext;
-        if (actionContextPlugin) {
-            actionContextPlugin(actionContext, self, self._app);
+    if (!self._actionContext) {
+        if (!self._dispatcher) {
+            self._initializeDispatcher();
         }
-    });
 
-    self._actionContext = actionContext;
+        var actionContext = {
+            dispatch: self._dispatcher.dispatch.bind(self._dispatcher),
+            executeAction: function actionExecuteAction (action, payload, callback) {
+                /*
+                 * Pass the current stack along.
+                 * If this action calls more actions,
+                 * we keep modifying and passing the stack along.
+                 */
+                var actionName = action.displayName || action.name;
+                // we have to clone the stack, can't just push to it
+                // since one action can execute multiple actions.
+                var nextStack = this.stack.concat([actionName]);
+                return self.executeAction(action, payload, callback, {
+                    nextStack: nextStack,
+                    id: this.id
+                });
+            },
+            getStore: self._dispatcher.getStore.bind(self._dispatcher)
+        };
+
+        self._plugins.forEach(function pluginsEach(plugin) {
+            var actionContextPlugin = plugin.plugActionContext;
+            if (actionContextPlugin) {
+                actionContextPlugin(actionContext, self, self._app);
+            }
+        });
+
+        self._actionContext = actionContext;
+    }
 
-    return self._actionContext;
+    var nextStack = (meta && meta.nextStack) || ['Action'];
```

Comment:
The first action will start with `Action` as the start of the stack trace. Usually this would be `navigateAction`